src/System.Management.Automation/cimSupport/cmdletization/ScriptWriter.cs

```diff
@@ -517,13 +517,56 @@ private Type GetDotNetType(TypeMetadata typeMetadata)
             Dbg.Assert(typeMetadata != null, "Caller should verify typeMetadata != null");
 
             string psTypeText;
-            List<EnumMetadataEnum> matchingEnums = (_cmdletizationMetadata.Enums ?? Enumerable.Empty<EnumMetadataEnum>())
-                .Where(e => Regex.IsMatch(
-                    typeMetadata.PSType,
-                    string.Format(CultureInfo.InvariantCulture, @"\b{0}\b", Regex.Escape(e.EnumName)),
-                    RegexOptions.CultureInvariant))
-                .ToList();
-            EnumMetadataEnum matchingEnum = matchingEnums.Count == 1 ? matchingEnums[0] : null;
+            EnumMetadataEnum matchingEnum = null;
+
+            if (_cmdletizationMetadata.Enums is not null)
+            {
+                string psType = typeMetadata.PSType;
+                foreach (EnumMetadataEnum e in _cmdletizationMetadata.Enums)
+                {
+                    int index = psType.IndexOf(e.EnumName, StringComparison.InvariantCulture);
```

Comment:
Maybe OrdinalIgnoreCase?

Reply:
I think InvariantCulture makes no sense for type names - it is linguistic search - we need exact equals.

Reply:
Since those are types, Ordinal (not IgnoreCase) sounds like the best fit.

Reply:
> Since those are types, Ordinal (not IgnoreCase) sounds like the best fit.

`psType` is not `Type`.

Reply:
Right. Those are strings that represent types. Correct? Hence may vary by case. Ordinal case-sensitive should work. Case-insensitive might not, and is more expensive.

Reply:
My concern is that this is not real reflection data.

Reply:
Although if InvariantCulture works, Ordinal should too.

Reply:
Although we don't use RegexOptions.IgnoreCase and Ordinal should work.

Reply:
Updated to use `Ordinal` instead.